### ISSUE TITLE
FB10185325: SwiftUI MenuBarExtra needs a style that does not present anything

### ISSUE THREAD

**Issue by sindresorhus** [feedback-assistant/reports#329]
- **Date:** 2022-06-12
- **Resolution:** Open
- **Area:** SwiftUI Framework
- **OS:** macOS 12.4
- **Type:** Suggestion

## Description

Right now, you can choose to either present a menu or a window when the user clicks on the menu bar item. However, there are many use-cases for wanting not to present anything automatically:

1. You want the menu bar item to simply be a toggle that toggles something.
2. You want to show your main app window when clicking the menu bar item.
3. You want to show a custom window that is not attached to the menu bar when clicking the menu bar item.

A specific example of number 3, is that clicking the menu bar item shows a live feed of your webcam which the user can drag around and place wherever they want on the screen.

Basically, I’d like a style where the menu bar item just works as a `Button` and let you do any action in the perform closure.
